rust/arrow/src/compute/kernels/arithmetic.rs

```diff
@@ -225,15 +205,7 @@ where
         unsafe { Buffer::try_from_trusted_len_iter(values) }
     }?;
 
-    let data = ArrayData::new(
```

Comment:
FWIW I think the new style of calling `ArrayData::new_primitive`  is much more understandable: Rather than 7 parameters which have some implicit assumptions about invariants between them, we have 2 parameters that make more sense.

Reply:
I agree!